### PR TITLE
feat(util actions): improve parsing for merge and parse actions

### PR DESCRIPTION
**Pull request by schultzp2020**
## Changes

- improve parsing for merge and parse actions
- improve importing of actions
- update README

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by schultzp2020** [comment]
I will add a changeset and update the tests later, I want to see what others think first.

**Comment by kissmikijr** [comment]
The aim of this basically to be able to just provide the path and we'll be able to figure out the parser based on the extension of the file, am I correct?

**Comment by schultzp2020** [comment]
> The aim of this basically to be able to just provide the path and we'll be able to figure out the parser based on the extension of the file, am I correct?

@kissmikijr Correct

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/utils/parse.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    return parsers[parser](content);
+  }
+
+  // Otherwise, try to guess based on file extension
+  switch (fileExtension) {
+    case '.json':
+      return parsers.json(content);
+    case '.yaml':
+    case '.yml':
+      return parsers.yaml(content);
```
Comment (punkle) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/utils/parse.ts:44]: Hmm. wondering if this should use to yaml or multiyaml? Thoughs.

**Comment by punkle** [comment]
This change needs a changeset.

**Pull request closed without merging**